modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/StoreQueryParser.java

```diff
@@ -273,39 +277,78 @@ private static StoreQueryRuntime constructStoreQueryRuntime(AggregationRuntime a
     }
 
     private static StoreQueryRuntime constructStoreQueryRuntime(Table table, StoreQuery storeQuery,
-                                                                SiddhiAppContext siddhiAppContext,
-                                                                Map<String, Table> tableMap, Map<String, Window>
+                                                             SiddhiAppContext siddhiAppContext,
+                                           Map<String, Table> tableMap, Map<String, Window> windowMap,
+                                           String queryName, int metaPosition, Expression onCondition,
+                                           MetaStreamEvent metaStreamEvent,
+                                           List<VariableExpressionExecutor> variableExpressionExecutors,
+                                           LockWrapper lockWrapper) {
+        if (table instanceof QueryableProcessor && storeQuery.getType() == StoreQuery.StoreQueryType.FIND) {
+            try {
+                return constructOptimizedStoreQueryRuntime(table, storeQuery, siddhiAppContext, tableMap,
+                        queryName, metaPosition, onCondition, metaStreamEvent, variableExpressionExecutors);
+            } catch (QueryableRecordTableException e) {
+                log.warn("Failed to create optimized Store Query Runtime as no query configuration was found " +
+                        "for table: " + table.getTableDefinition().getId());
```

Comment:
Move this to info log, as this is something the user does not need to worry about. 
Also, add a reason why we are not optimizing in the debug log.